cpp/src/glm/qn/qn_solvers.cuh

```diff
@@ -235,9 +235,9 @@ inline OPT_RETCODE min_owlqn(const LBFGSParam<T> &param, Function &f,
 
   op_project<T> project_neg(T(-1.0));
 
-  auto f_wrap = [&f, &l1_penalty, &pg_limit, &stream](
-                  SimpleVec<T> &x, SimpleVec<T> &grad, T *dev_scalar,
-                  cudaStream_t stream) {
+  auto f_wrap = [&f, &l1_penalty, &pg_limit](SimpleVec<T> &x,
+                                             SimpleVec<T> &grad, T *dev_scalar,
+                                             cudaStream_t stream) {
```

Comment:
Given that the lambda was capturing `stream` by reference and taking the argument `stream` by value, I wanted to double-check that we do indeed want to take the `stream` argument by value here? Other than that, looks great!

Reply:
That's a good question! `stream` by value, I believe, is also just a typedef of an address (so, a pointer) so we should be fine passing it by reference or by value

Reply:
In this case, we have a lambda function which accepts a stream as an argument, performs some processing, and propagates the stream further through other function calls. This is the way it should be done- rather than using the explicit stream from the environment in which the lambda was created, it should be getting propagated through the function call stack.